functions-python/feed_sync_dispatcher_transitland/src/main.py

```diff
@@ -0,0 +1,305 @@
+
+    def to_dict(self):
+        return asdict(self)
+
+    def to_json(self):
+        return json.dumps(self.to_dict())
+
+
+class TransitFeedSyncProcessor(FeedSyncProcessor):
+    def check_url_status(self, url: str) -> bool:
+        """
+        Checks if a URL returns a valid response (not 404 or 500).
+        """
+        try:
+            response = requests.head(url, timeout=10)
+            return response.status_code not in {404, 500}
+        except requests.RequestException as e:
+            logging.warning(f"Failed to reach {url}: {e}")
+            return False
+
+    def process_sync(self, db_session: Optional[Session] = None, execution_id: Optional[str] = None) -> List[FeedSyncPayload]:
+        """
+        Process data synchronously to fetch, extract, combine, filter and prepare payloads for publishing
+        to a queue based on conditions related to the data retrieved from TransitLand API.
+        """
+        feeds_data = self.get_data(Tlnd_feeds_endpoint, apikey, spec, session)
+        operators_data = self.get_data(Tlnd_operators_endpoint, apikey, session=session)
+
+        feeds = self.extract_feeds_data(feeds_data)
+        operators = self.extract_operators_data(operators_data)
+
+        # Converts operators and feeds to pandas DataFrames
+        operators_df = pd.DataFrame(operators)
+        feeds_df = pd.DataFrame(feeds)
+
+        # Merge operators and feeds DataFrames on 'operator_feed_id' and 'feed_id'
+        combined_df = pd.merge(operators_df, feeds_df, left_on='operator_feed_id', right_on='feed_id', how='inner')
+
+        # Filter out rows where 'feed_url' is missing
+        combined_df = combined_df[combined_df['feed_url'].notna()]
+
+        # Group by 'feed_id' and concatenate 'operator_name' while keeping first values of other columns
+        df_grouped = combined_df.groupby('feed_id').agg({
+            'operator_name': lambda x: ', '.join(x),
+            'feeds_onestop_id': 'first',
+            'feed_url': 'first',
+            'operator_feed_id': 'first',
+            'spec': 'first',
+            'country': 'first',
+            'state_province': 'first',
+            'city_name': 'first',
+            'auth_info_url': 'first',
+            'auth_param_name': 'first',
+            'type': 'first'
+        }).reset_index()
+
+        # Filter out unwanted countries
+        countries_not_included = ['France', 'Japan']
+        filtered_df = df_grouped[~df_grouped['country'].str.lower().isin([c.lower() for c in countries_not_included])]
+
+        #  Filter out unwanted response_code
+        filtered_df = filtered_df[filtered_df['feed_url'].apply(self.check_url_status)]
+
+        # Converts filtered DataFrame to dictionary format
+        combined_data = filtered_df.to_dict(orient='records')
+
+        # payloads for publishing to queue
+        payloads = []
+        for data in combined_data:
+            associated_id = self.get_associated_id(db_session, data['feeds_onestop_id'])
```

Comment:
We need to filter the associated ID by the ID and the source. This ensures that the ID doesn't conflict with any other source. You can use the transit land source as: _"TLD"_.